doc/source/developer.rst

```diff
@@ -50,15 +50,36 @@ So that a ``pandas.DataFrame`` can be faithfully reconstructed, we store a
     'pandas_version': $VERSION}
 
 Here, ``<c0>``/``<ci0>`` and so forth are dictionaries containing the metadata
-for each column. This has JSON form:
+for each column, *including the index columns*. This has JSON form:
 
 .. code-block:: text
 
    {'name': column_name,
+    'field_name': arrow_table_column_name,
```

Comment:
arrow_table_column_name -> parquet_(schema)/(column)_name ?

Reply:
I don't think so, since this isn't about parquet it's about pandas metadata inside of arrow objects.

Reply:
It is also relevant for `fastparquet` no? I mean, ideally they should also add this field to the metadata, and it should correspond to the names they use in the parquet file (which will also be different for indexes without a name)

Reply:
Hm fair enough. I will change.